model/pipeline/src/main/resources/org/apache/beam/model/common_urns.md

```diff
@@ -0,0 +1,122 @@
+<!--
+
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# Apache Beam URNs
+
+This file serves as a central place to enumerate and document the various
+URNs used in the Beam portability APIs.
+
+
+## Core Transforms
+
+### org.apache.beam:transform:pardo:v1
```

Comment:
<!--new_thread; commit:8d3085bf4a646efa65e3a40f0c4528297fb54694; resolved:0-->
Technically, a URN has the form `urn:` _<namespace>_ `:` _<namespace-specific section>_

So, we drop the `urn` since it is redundant from context and I'm not sure we really need `org.apache.beam` as the namespace. It isn't an "authority" section as in other URIs. We can just use `beam` as we already have in most of the constants. The actual process is just that you get it registered with IANA [here](https://www.iana.org/assignments/urn-namespaces/urn-namespaces.xhtml) but I don't know that we need to. TBH even "beam" is probably redundant when we are putting them into URN fields in a Beam proto.

Reply:
Should we follow semantic versioning for the version numbers similar to how we are releasing the SDK versions (e.g. org.apache.beam:transform:pardo:1.0.0)?

Reply:
Currently we're just requiring exact matches. I'm not sure what a "bugfix" version means for URNs, I suppose minor upgrades could mean adding formerly optional fields to proto-like payloads, but that could be checked for directly as well.

Reply:
It is pretty common for protocols to use the minor version, but I don't know of any that use the bugfix version.

There's a polarity issue in the higher-order case that seems to also apply here - adding a field to an interface is backwards compatible for consumers, but incompatible for implementers of the interface. But adding an optional field is both backward and forward compatible so you don't need a minor version bump.